Add NanoPi R2C support

rickliu2000 opened the pull request:
# Description
Add support for NanoPi R2C. The board is identical to NanoPi R2S but with a YT8521s PHY instead of the realtek one.
The kernel and uboot are the same as R2S with extra patch to support the PHY.


# How Has This Been Tested?

I have built an image for the NanoPi R2C locally, it boots and worked fine. Both of the Ethernet ports are working.

dmesg: https://pastebin.com/yMM5gyrK



# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules

igorpecovnik commented:
> I have built an image for the NanoPi R2C locally, it boots and worked fine. Both of the Ethernet ports are working.

Can you also try with EDGE kernel 5.14.y. ?

rickliu2000 commented:
> > I have built an image for the NanoPi R2C locally, it boots and worked fine. Both of the Ethernet ports are working.
> 
> Can you also try with EDGE kernel 5.14.y. ?

The motorcom phy dirver won't build at 5.14, also there seems to be no R2S related patch in current 5.14 patch folder.

rickliu2000 commented:
> > I have built an image for the NanoPi R2C locally, it boots and worked fine. Both of the Ethernet ports are working.
> 
> Can you also try with EDGE kernel 5.14.y. ?

Just fixed the driver and built an image on edge, unfortunately, it doesn't boot. Sorry I don't have more time to investigate further on this.

rickliu2000 commented:
> > I have built an image for the NanoPi R2C locally, it boots and worked fine. Both of the Ethernet ports are working.
> 
> Can you also try with EDGE kernel 5.14.y. ?

It worked now, and I assume R2S will also work with my commit.

dmesg:
[https://pastebin.com/Dz3Yn4Pq](url)

igorpecovnik commented:
> Edge works now, and I assume R2S will also work with my commit.

Yes, it looks ok. Thank you.

MichaIng commented:
@rickliu2000 
Is the YT8521S adapter still supposed to work? We have a user where it didn't:
```
[ 8.405453] rk_gmac-dwmac ff540000.ethernet eth0: no phy at addr -1
[ 8.406040] rk_gmac-dwmac ff540000.ethernet eth0: stmmac_open: Cannot attach to PHY (error: -19)
```
We tested "current" (Linux 5.10) R2S Bullseye image, then after flashing the R2C "current" U-Boot image, then tried "edge" (Linux 5.15) kernel, dtb and U-Boot, with same result, using also the `armbian-firmware` package. The LAN adapter works well, but the WAN adapter fails. Let me know if we can help debugging. I know there is no maintainer for this SBC, so no claim ✌️.

igorpecovnik commented:
Possibly [this](https://github.com/armbian/build/pull/3514) broke it.

MichaIng commented:
Hmm, the issue showed up before this PR was even created. But still, two huge patches to the same driver, taken unmodified from different downstream providers, is a dangerous thing when not carefully reviewing for conflicts or testing intensively on all devices 😉.

schwar3kat commented:
> Hmm, the issue showed up before this PR was even created.

@MichaIng - sounds like this second patch might not be the issue, but if you want to make sure, then we need to test without the second patch.  The YT8521S is theoretically in the first patch.  I'm assuming that it worked before?  The user is using Diet Pi.  I'm not familiar with your build process.  Are you able to do a build without the second patch?

rickliu2000 commented:
> @rickliu2000 Is the YT8521S adapter still supposed to work? We have a user where it didn't:
> 
> ```
> [ 8.405453] rk_gmac-dwmac ff540000.ethernet eth0: no phy at addr -1
> [ 8.406040] rk_gmac-dwmac ff540000.ethernet eth0: stmmac_open: Cannot attach to PHY (error: -19)
> ```
> 
> We tested "current" (Linux 5.10) R2S Bullseye image, then after flashing the R2C "current" U-Boot image, then tried "edge" (Linux 5.15) kernel, dtb and U-Boot, with same result, using also the `armbian-firmware` package. The LAN adapter works well, but the WAN adapter fails. Let me know if we can help debugging. I know there is no maintainer for this SBC, so no claim ✌️.

Hi, and Hi to all in this thread. After a couple of years I think I have some time to look into the issue again. Seems we now have a yt8521 phy driver in the mainline linux kernel now. See drivers/net/phy/motorcomm.c. We may be able to enable this phy easier than before without the need for that driver patch.

rickliu2000 commented:
@igorpecovnik  Hi I can confirm that both of the ethernet port of nanopi r2c work with the device tree rk3328-nanopi-r2-rev06.dts 
The device tree from mainline linux ending with r2c seems not to be working.
Also I noticed that I missed some discussion about maintaing this device, don't know if there is anything I can do to let the nanopi r2c image generated by armbian. 
`[    0.000000] Booting Linux on physical CPU 0x0000000000 [0x410fd034]
[    0.000000] Linux version 6.12.10-current-rockchip64 (build@armbian) (aarch64-linux-gnu-gcc (Ubuntu 11.4.0-1ubuntu1~22.04) 11.4.0, GNU ld (GNU Binutils for Ubuntu) 2.38) #2 SMP PREEMPT Fri Jan 17 12:41:00 UTC 2025
[    0.000000] KASLR disabled due to lack of seed
[    0.000000] Machine model: FriendlyElec NanoPi R2C
[    0.000000] efi: UEFI not found.
[    0.000000] NUMA: Faking a node at [mem 0x0000000000200000-0x000000003fffffff]
[    0.000000] NODE_DATA(0) allocated [mem 0x3fdb4240-0x3fdb69bf]
[    0.000000] Zone ranges:
[    0.000000]   DMA      [mem 0x0000000000200000-0x000000003fffffff]
[    0.000000]   DMA32    empty
[    0.000000]   Normal   empty
[    0.000000] Movable zone start for each node
[    0.000000] Early memory node ranges
[    0.000000]   node   0: [mem 0x0000000000200000-0x000000003fffffff]
[    0.000000] Initmem setup node 0 [mem 0x0000000000200000-0x000000003fffffff]
[    0.000000] On node 0, zone DMA: 512 pages in unavailable ranges
[    0.000000] cma: Reserved 128 MiB at 0x0000000034a00000 on node -1
[    0.000000] psci: probing for conduit method from DT.
[    0.000000] psci: PSCIv1.0 detected in firmware.
[    0.000000] psci: Using standard PSCI v0.2 function IDs
[    0.000000] psci: MIGRATE_INFO_TYPE not supported.
[    0.000000] psci: SMC Calling Convention v1.0
[    0.000000] percpu: Embedded 34 pages/cpu s98968 r8192 d32104 u139264
[    0.000000] pcpu-alloc: s98968 r8192 d32104 u139264 alloc=34*4096
[    0.000000] pcpu-alloc: [0] 0 [0] 1 [0] 2 [0] 3
[    0.000000] Detected VIPT I-cache on CPU0
[    0.000000] CPU features: detected: ARM erratum 845719
[    0.000000] alternatives: applying boot alternatives
[    0.000000] Kernel command line: root=UUID=ed18e1a3-fb1e-4044-9e49-1397fba1aa37 rootwait rootfstype=ext4 splash=verbose console=ttyS2,1500000  consoleblank=0 loglevel=1 ubootpart=5ee3a786-01 usb-storage.quirks=0x2537:0x1066:u,0x2537:0x1068:u   cgroup_enable=cpuset cgroup_memory=1 cgroup_enable=memory swapaccount=1
[    0.000000] Unknown kernel command line parameters "splash=verbose ubootpart=5ee3a786-01 cgroup_enable=memory cgroup_memory=1", will be passed to user space.
[    0.000000] Dentry cache hash table entries: 131072 (order: 8, 1048576 bytes, linear)
[    0.000000] Inode-cache hash table entries: 65536 (order: 7, 524288 bytes, linear)
[    0.000000] Fallback order for Node 0: 0
[    0.000000] Built 1 zonelists, mobility grouping on.  Total pages: 261632
[    0.000000] Policy zone: DMA
[    0.000000] mem auto-init: stack:off, heap alloc:on, heap free:off
[    0.000000] software IO TLB: SWIOTLB bounce buffer size adjusted to 0MB
[    0.000000] software IO TLB: area num 4.
[    0.000000] software IO TLB: mapped [mem 0x000000003ea80000-0x000000003eb80000] (1MB)
[    0.000000] SLUB: HWalign=64, Order=0-3, MinObjects=0, CPUs=4, Nodes=1
[    0.000000] rcu: Preemptible hierarchical RCU implementation.
[    0.000000] rcu:     RCU event tracing is enabled.
[    0.000000] rcu:     RCU restricting CPUs from NR_CPUS=256 to nr_cpu_ids=4.
[    0.000000]  Trampoline variant of Tasks RCU enabled.
[    0.000000]  Tracing variant of Tasks RCU enabled.
[    0.000000] rcu: RCU calculated value of scheduler-enlistment delay is 25 jiffies.
[    0.000000] rcu: Adjusting geometry for rcu_fanout_leaf=16, nr_cpu_ids=4
[    0.000000] RCU Tasks: Setting shift to 2 and lim to 1 rcu_task_cb_adjust=1 rcu_task_cpu_ids=4.
[    0.000000] RCU Tasks Trace: Setting shift to 2 and lim to 1 rcu_task_cb_adjust=1 rcu_task_cpu_ids=4.
[    0.000000] NR_IRQS: 64, nr_irqs: 64, preallocated irqs: 0
[    0.000000] Root IRQ handler: gic_handle_irq
[    0.000000] GIC: Using split EOI/Deactivate mode
[    0.000000] rcu: srcu_init: Setting srcu_struct sizes based on contention.
[    0.000000] arch_timer: cp15 timer(s) running at 24.00MHz (phys).
[    0.000000] clocksource: arch_sys_counter: mask: 0xffffffffffffff max_cycles: 0x588fe9dc0, max_idle_ns: 440795202592 ns
[    0.000001] sched_clock: 56 bits at 24MHz, resolution 41ns, wraps every 4398046511097ns
[    0.001133] Console: colour dummy device 80x25
[    0.001329] Calibrating delay loop (skipped), value calculated using timer frequency.. 48.00 BogoMIPS (lpj=96000)
[    0.001354] pid_max: default: 32768 minimum: 301
[    0.001534] LSM: initializing lsm=capability,yama,apparmor
[    0.001639] Yama: becoming mindful.
[    0.002039] AppArmor: AppArmor initialized
[    0.002229] Mount-cache hash table entries: 2048 (order: 2, 16384 bytes, linear)
[    0.002256] Mountpoint-cache hash table entries: 2048 (order: 2, 16384 bytes, linear)
[    0.027919] rcu: Hierarchical SRCU implementation.
[    0.027938] rcu:     Max phase no-delay instances is 1000.
[    0.028512] Timer migration: 1 hierarchy levels; 8 children per group; 1 crossnode level
[    0.030161] EFI services will not be available.
[    0.030911] smp: Bringing up secondary CPUs ...
[    0.032155] Detected VIPT I-cache on CPU1
[    0.032361] CPU1: Booted secondary processor 0x0000000001 [0x410fd034]
[    0.033969] Detected VIPT I-cache on CPU2
[    0.034171] CPU2: Booted secondary processor 0x0000000002 [0x410fd034]
[    0.040451] Detected VIPT I-cache on CPU3
[    0.040663] CPU3: Booted secondary processor 0x0000000003 [0x410fd034]
[    0.040924] smp: Brought up 1 node, 4 CPUs
[    0.040942] SMP: Total of 4 processors activated.
[    0.040949] CPU: All CPU(s) started at EL2
[    0.041005] CPU features: detected: 32-bit EL0 Support
[    0.041010] CPU features: detected: 32-bit EL1 Support
[    0.041022] CPU features: detected: CRC32 instructions
[    0.041116] alternatives: applying system-wide alternatives
[    0.043274] Memory: 832244K/1046528K available (16384K kernel code, 2914K rwdata, 11072K rodata, 5120K init, 786K bss, 78508K reserved, 131072K cma-reserved)
[    0.044978] devtmpfs: initialized
[    0.067652] clocksource: jiffies: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 7645041785100000 ns
[    0.067724] futex hash table entries: 1024 (order: 4, 65536 bytes, linear)
[    0.075352] 23648 pages in range for non-PLT usage
[    0.075379] 515168 pages in range for PLT usage
[    0.075920] pinctrl core: initialized pinctrl subsystem
[    0.076972] DMI not present or invalid.
[    0.082643] NET: Registered PF_NETLINK/PF_ROUTE protocol family
[    0.086474] DMA: preallocated 2048 KiB GFP_KERNEL pool for atomic allocations
[    0.089230] DMA: preallocated 2048 KiB GFP_KERNEL|GFP_DMA pool for atomic allocations
[    0.092529] DMA: preallocated 2048 KiB GFP_KERNEL|GFP_DMA32 pool for atomic allocations
[    0.093550] audit: initializing netlink subsys (disabled)
[    0.094197] audit: type=2000 audit(0.092:1): state=initialized audit_enabled=0 res=1
[    0.095740] thermal_sys: Registered thermal governor 'fair_share'
[    0.095751] thermal_sys: Registered thermal governor 'bang_bang'
[    0.095762] thermal_sys: Registered thermal governor 'step_wise'
[    0.095768] thermal_sys: Registered thermal governor 'user_space'
[    0.095931] cpuidle: using governor menu
[    0.096500] hw-breakpoint: found 6 breakpoint and 4 watchpoint registers.
[    0.096760] ASID allocator initialised with 65536 entries
[    0.097273] Serial: AMBA PL011 UART driver
[    0.124087] gpio gpiochip0: Static allocation of GPIO base is deprecated, use dynamic allocation.
[    0.125011] rockchip-gpio ff210000.gpio: probed /pinctrl/gpio@ff210000
[    0.125967] gpio gpiochip1: Static allocation of GPIO base is deprecated, use dynamic allocation.
[    0.126628] rockchip-gpio ff220000.gpio: probed /pinctrl/gpio@ff220000
[    0.127539] gpio gpiochip2: Static allocation of GPIO base is deprecated, use dynamic allocation.
[    0.128123] rockchip-gpio ff230000.gpio: probed /pinctrl/gpio@ff230000
[    0.128855] gpio gpiochip3: Static allocation of GPIO base is deprecated, use dynamic allocation.
[    0.129466] rockchip-gpio ff240000.gpio: probed /pinctrl/gpio@ff240000
[    0.137790] HugeTLB: registered 1.00 GiB page size, pre-allocated 0 pages
[    0.137813] HugeTLB: 0 KiB vmemmap can be freed for a 1.00 GiB page
[    0.137824] HugeTLB: registered 32.0 MiB page size, pre-allocated 0 pages
[    0.137833] HugeTLB: 0 KiB vmemmap can be freed for a 32.0 MiB page
[    0.137844] HugeTLB: registered 2.00 MiB page size, pre-allocated 0 pages
[    0.137853] HugeTLB: 0 KiB vmemmap can be freed for a 2.00 MiB page
[    0.137864] HugeTLB: registered 64.0 KiB page size, pre-allocated 0 pages
[    0.137874] HugeTLB: 0 KiB vmemmap can be freed for a 64.0 KiB page
[    0.140561] cryptd: max_cpu_qlen set to 1000
[    0.207450] raid6: neonx8   gen()  1071 MB/s
[    0.275617] raid6: neonx4   gen()  1095 MB/s
[    0.343791] raid6: neonx2   gen()  1029 MB/s
[    0.411907] raid6: neonx1   gen()   874 MB/s
[    0.480053] raid6: int64x8  gen()   682 MB/s
[    0.548191] raid6: int64x4  gen()   797 MB/s
[    0.616341] raid6: int64x2  gen()   699 MB/s
[    0.684460] raid6: int64x1  gen()   515 MB/s
[    0.684472] raid6: using algorithm neonx4 gen() 1095 MB/s
[    0.752601] raid6: .... xor() 778 MB/s, rmw enabled
[    0.752610] raid6: using neon recovery algorithm
[    0.754003] iommu: Default domain type: Translated
[    0.754027] iommu: DMA domain TLB invalidation policy: strict mode
[    0.756779] SCSI subsystem initialized
[    0.757280] libata version 3.00 loaded.
[    0.757807] usbcore: registered new interface driver usbfs
[    0.757863] usbcore: registered new interface driver hub
[    0.757956] usbcore: registered new device driver usb
[    0.758601] pps_core: LinuxPPS API ver. 1 registered
[    0.758611] pps_core: Software ver. 5.3.6 - Copyright 2005-2007 Rodolfo Giometti <giometti@linux.it>
[    0.758641] PTP clock support registered
[    0.758715] EDAC MC: Ver: 3.0.0
[    0.759502] scmi_core: SCMI protocol bus registered
[    0.761860] NetLabel: Initializing
[    0.761877] NetLabel:  domain hash size = 128
[    0.761884] NetLabel:  protocols = UNLABELED CIPSOv4 CALIPSO
[    0.762048] NetLabel:  unlabeled traffic allowed by default
[    0.762460] vgaarb: loaded
[    0.763502] clocksource: Switched to clocksource arch_sys_counter
[    0.770358] VFS: Disk quotas dquot_6.6.0
[    0.770431] VFS: Dquot-cache hash table entries: 512 (order 0, 4096 bytes)
[    0.771714] AppArmor: AppArmor Filesystem Enabled
[    0.788991] NET: Registered PF_INET protocol family
[    0.789399] IP idents hash table entries: 16384 (order: 5, 131072 bytes, linear)
[    0.909451] tcp_listen_portaddr_hash hash table entries: 512 (order: 1, 8192 bytes, linear)
[    0.909628] Table-perturb hash table entries: 65536 (order: 6, 262144 bytes, linear)
[    0.909806] TCP established hash table entries: 8192 (order: 4, 65536 bytes, linear)
[    0.910037] TCP bind hash table entries: 8192 (order: 6, 262144 bytes, linear)
[    0.910433] TCP: Hash tables configured (established 8192 bind 8192)
[    0.910697] UDP hash table entries: 512 (order: 2, 16384 bytes, linear)
[    0.910767] UDP-Lite hash table entries: 512 (order: 2, 16384 bytes, linear)
[    0.911194] NET: Registered PF_UNIX/PF_LOCAL protocol family
[    0.911252] NET: Registered PF_XDP protocol family
[    0.911278] PCI: CLS 0 bytes, default 64
[    0.911969] Trying to unpack rootfs image as initramfs...
[    0.921607] kvm [1]: nv: 554 coarse grained trap handlers
[    0.922584] kvm [1]: IPA Size Limit: 40 bits
[    0.927594] kvm [1]: vgic interrupt IRQ9
[    0.927677] kvm [1]: Hyp nVHE mode initialized successfully
[    0.931932] Initialise system trusted keyrings
[    0.932066] Key type blacklist registered
[    0.932638] workingset: timestamp_bits=44 max_order=18 bucket_order=0
[    0.932742] zbud: loaded
[    0.933818] squashfs: version 4.0 (2009/01/31) Phillip Lougher
[    0.934888] fuse: init (API version 7.41)
[    0.938351] integrity: Platform Keyring initialized
[    1.046726] xor: measuring software checksum speed
[    1.049444]    8regs           :  1219 MB/sec
[    1.052131]    32regs          :  1234 MB/sec
[    1.054941]    arm64_neon      :  1178 MB/sec
[    1.054960] xor: using function: 32regs (1234 MB/sec)
[    1.054990] Key type asymmetric registered
[    1.055002] Asymmetric key parser 'x509' registered
[    1.055288] Block layer SCSI generic (bsg) driver version 0.4 loaded (major 246)
[    1.055924] io scheduler mq-deadline registered
[    1.055948] io scheduler kyber registered
[    1.056078] io scheduler bfq registered
[    3.114023] Freeing initrd memory: 20080K
[    3.119626] ledtrig-cpu: registered to indicate activity on CPUs
[    3.126004] dma-pl330 ff1f0000.dma-controller: Loaded driver for PL330 DMAC-241330
[    3.126036] dma-pl330 ff1f0000.dma-controller:       DBUFF-128x8bytes Num_Chans-8 Num_Peri-20 Num_Events-16
[    3.128981] Serial: 8250/16550 driver, 8 ports, IRQ sharing disabled
[    3.137710] printk: legacy console [ttyS2] disabled
[    3.138541] ff130000.serial: ttyS2 at MMIO 0xff130000 (irq = 23, base_baud = 1500000) is a 16550A
[    3.138713] printk: legacy console [ttyS2] enabled
[    3.141116] Serial: AMBA driver
[    3.160311] loop: module loaded
[    3.162549] Key type psk registered
[    3.169221] thunder_xcv, ver 1.0
[    3.169314] thunder_bgx, ver 1.0
[    3.169375] nicpf, ver 1.0
[    3.173119] dwc2 ff580000.usb: supply vusb_d not found, using dummy regulator
[    3.173383] dwc2 ff580000.usb: supply vusb_a not found, using dummy regulator
[    3.299648] dwc2 ff580000.usb: EPs: 10, dedicated fifos, 972 entries in SPRAM
[    3.300056] dwc2 ff580000.usb: DWC OTG Controller
[    3.300125] dwc2 ff580000.usb: new USB bus registered, assigned bus number 1
[    3.300191] dwc2 ff580000.usb: irq 28, io mem 0xff580000
[    3.300610] usb usb1: New USB device found, idVendor=1d6b, idProduct=0002, bcdDevice= 6.12
[    3.300636] usb usb1: New USB device strings: Mfr=3, Product=2, SerialNumber=1
[    3.300655] usb usb1: Product: DWC OTG Controller
[    3.300669] usb usb1: Manufacturer: Linux 6.12.10-current-rockchip64 dwc2_hsotg
[    3.300686] usb usb1: SerialNumber: ff580000.usb
[    3.301878] hub 1-0:1.0: USB hub found
[    3.301957] hub 1-0:1.0: 1 port detected
[    3.305298] xhci-hcd xhci-hcd.0.auto: xHCI Host Controller
[    3.305367] xhci-hcd xhci-hcd.0.auto: new USB bus registered, assigned bus number 2
[    3.305654] xhci-hcd xhci-hcd.0.auto: hcc params 0x0220fe64 hci version 0x110 quirks 0x0000808002000010
[    3.305774] xhci-hcd xhci-hcd.0.auto: irq 27, io mem 0xff600000
[    3.306113] xhci-hcd xhci-hcd.0.auto: xHCI Host Controller
[    3.306142] xhci-hcd xhci-hcd.0.auto: new USB bus registered, assigned bus number 3
[    3.306175] xhci-hcd xhci-hcd.0.auto: Host supports USB 3.0 SuperSpeed
[    3.306576] usb usb2: New USB device found, idVendor=1d6b, idProduct=0002, bcdDevice= 6.12
[    3.306603] usb usb2: New USB device strings: Mfr=3, Product=2, SerialNumber=1
[    3.306628] usb usb2: Product: xHCI Host Controller
[    3.306645] usb usb2: Manufacturer: Linux 6.12.10-current-rockchip64 xhci-hcd
[    3.306662] usb usb2: SerialNumber: xhci-hcd.0.auto
[    3.307345] ehci-platform ff5c0000.usb: EHCI Host Controller
[    3.307373] ohci-platform ff5d0000.usb: Generic Platform OHCI controller
[    3.308128] hub 2-0:1.0: USB hub found
[    3.308207] hub 2-0:1.0: 1 port detected
[    3.308987] usb usb3: We don't know the algorithms for LPM for this host, disabling LPM.
[    3.309259] usb usb3: New USB device found, idVendor=1d6b, idProduct=0003, bcdDevice= 6.12
[    3.309283] usb usb3: New USB device strings: Mfr=3, Product=2, SerialNumber=1
[    3.309303] usb usb3: Product: xHCI Host Controller
[    3.309319] usb usb3: Manufacturer: Linux 6.12.10-current-rockchip64 xhci-hcd
[    3.309339] usb usb3: SerialNumber: xhci-hcd.0.auto
[    3.310474] hub 3-0:1.0: USB hub found
[    3.310572] hub 3-0:1.0: 1 port detected
[    3.311308] ehci-platform ff5c0000.usb: new USB bus registered, assigned bus number 4
[    3.311335] ohci-platform ff5d0000.usb: new USB bus registered, assigned bus number 5
[    3.311578] usbcore: registered new interface driver usb-storage
[    3.311678] ehci-platform ff5c0000.usb: irq 29, io mem 0xff5c0000
[    3.311714] ohci-platform ff5d0000.usb: irq 30, io mem 0xff5d0000
[    3.314075] mousedev: PS/2 mouse device common for all mice
[    3.314874] i2c_dev: i2c /dev entries driver
[    3.318005] /i2c@ff160000/rk805@18: Fixed dependency cycle(s) with /i2c@ff160000/rk805@18/regulators/DCDC_REG4
[    3.327582] ehci-platform ff5c0000.usb: USB 2.0 started, EHCI 1.00
[    3.328086] usb usb4: New USB device found, idVendor=1d6b, idProduct=0002, bcdDevice= 6.12
[    3.328120] usb usb4: New USB device strings: Mfr=3, Product=2, SerialNumber=1
[    3.328165] usb usb4: Product: EHCI Host Controller
[    3.328180] usb usb4: Manufacturer: Linux 6.12.10-current-rockchip64 ehci_hcd
[    3.328194] usb usb4: SerialNumber: ff5c0000.usb
[    3.329365] hub 4-0:1.0: USB hub found
[    3.329458] hub 4-0:1.0: 1 port detected
[    3.350047] rockchip-thermal ff250000.tsadc: Missing tshut mode property, using default (cru)
[    3.350092] rockchip-thermal ff250000.tsadc: Missing tshut-polarity property, using default (low)
[    3.352491] dw_wdt ff1a0000.watchdog: No valid TOPs array specified
[    3.355831] sdhci: Secure Digital Host Controller Interface driver
[    3.355864] sdhci: Copyright(c) Pierre Ossman
[    3.355925] Synopsys Designware Multimedia Card Interface Driver
[    3.356827] sdhci-pltfm: SDHCI platform and OF driver helper
[    3.358527] hid: raw HID events driver (C) Jiri Kosina
[    3.358731] usbcore: registered new interface driver usbhid
[    3.358746] usbhid: USB HID core driver
[    3.360972] rockchip-dfi ff790000.dfi: dfi initialized, dram type: 0x0, channels: 1
[    3.363708] hw perfevents: enabled with armv8_cortex_a53 PMU driver, 7 (0,8000003f) counters available
[    3.365777] NET: Registered PF_INET6 protocol family
[    3.372163] usb usb5: New USB device found, idVendor=1d6b, idProduct=0001, bcdDevice= 6.12
[    3.372216] usb usb5: New USB device strings: Mfr=3, Product=2, SerialNumber=1
[    3.372234] usb usb5: Product: Generic Platform OHCI controller
[    3.372249] usb usb5: Manufacturer: Linux 6.12.10-current-rockchip64 ohci_hcd
[    3.372263] usb usb5: SerialNumber: ff5d0000.usb
[    3.373895] hub 5-0:1.0: USB hub found
[    3.374025] hub 5-0:1.0: 1 port detected
[    3.454623] Segment Routing with IPv6
[    3.454834] In-situ OAM (IOAM) with IPv6
[    3.455017] NET: Registered PF_PACKET protocol family
[    3.455590] 8021q: 802.1Q VLAN Support v1.8
[    3.455734] Key type dns_resolver registered
[    3.476663] registered taskstats version 1
[    3.477375] Loading compiled-in X.509 certificates
[    3.507935] zswap: loaded using pool zstd/z3fold
[    3.508703] Demotion targets for Node 0: null
[    3.509879] Key type .fscrypt registered
[    3.509911] Key type fscrypt-provisioning registered
[    3.515698] Btrfs loaded, zoned=yes, fsverity=yes
[    3.516160] Key type encrypted registered
[    3.516196] AppArmor: AppArmor sha256 policy hashing enabled
[    3.550308] rk3328-dmc ff780000.dmc: current ATF version 0x101
[    3.551598] dwmmc_rockchip ff500000.mmc: IDMAC supports 32-bit address mode.
[    3.551688] dwmmc_rockchip ff500000.mmc: Using internal DMA controller.
[    3.551708] dwmmc_rockchip ff500000.mmc: Version ID is 270a
[    3.551816] dwmmc_rockchip ff500000.mmc: DW MMC controller at irq 48,32 bit host data width,256 deep fifo
[    3.570909] mmc_host mmc0: Bus speed (slot 0) = 400000Hz (slot req 400000Hz, actual 400000HZ div = 0)
[    3.584319] WCN: marlin_init entry!
[    3.585798] input: gpio-keys as /devices/platform/gpio-keys/input/input0
[    3.587087] of_cfs_init
[    3.587179] of_cfs_init: OK
[    3.646871] mmc_host mmc0: Bus speed (slot 0) = 150000000Hz (slot req 150000000Hz, actual 150000000HZ div = 0)
[    3.740592] clk: Disabling unused clocks
[    3.741528] PM: genpd: Disabling unused power domains
[    3.741840] dw-apb-uart ff130000.serial: forbid DMA for kernel console
[    3.744656] Freeing unused kernel memory: 5120K
[    3.744951] Run /init as init process
[    3.744967]   with arguments:
[    3.744978]     /init
[    3.744989]   with environment:
[    3.744997]     HOME=/
[    3.745007]     TERM=linux
[    3.745016]     splash=verbose
[    3.745026]     ubootpart=5ee3a786-01
[    3.745036]     cgroup_enable=memory
[    3.745046]     cgroup_memory=1
[    3.782071] dwmmc_rockchip ff500000.mmc: Successfully tuned phase to 184
[    3.782127] mmc0: new ultra high speed SDR104 SDXC card at address 59b4
[    3.787727] mmcblk0: mmc0:59b4 SD64G 59.4 GiB
[    3.791770]  mmcblk0: p1
[    4.464512] rk808-rtc rk808-rtc.4.auto: registered as rtc0
[    4.466415] rk808-rtc rk808-rtc.4.auto: setting system clock to 2025-01-17T14:48:48 UTC (1737125328)
[    4.561204] rk_gmac-dwmac ff540000.ethernet: IRQ eth_wake_irq not found
[    4.561239] rk_gmac-dwmac ff540000.ethernet: IRQ eth_lpi not found
[    4.561254] rk_gmac-dwmac ff540000.ethernet: IRQ sfty not found
[    4.561478] rk_gmac-dwmac ff540000.ethernet: PTP uses main clock
[    4.561916] rk_gmac-dwmac ff540000.ethernet: clock input or output? (input).
[    4.561950] rk_gmac-dwmac ff540000.ethernet: TX delay(0x22).
[    4.561968] rk_gmac-dwmac ff540000.ethernet: RX delay(0x12).
[    4.561996] rk_gmac-dwmac ff540000.ethernet: integrated PHY? (no).
[    4.562071] rk_gmac-dwmac ff540000.ethernet: clock input from PHY
[    4.567105] rk_gmac-dwmac ff540000.ethernet: init for RGMII
[    4.567680] rk_gmac-dwmac ff540000.ethernet: User ID: 0x10, Synopsys ID: 0x35
[    4.567727] rk_gmac-dwmac ff540000.ethernet:         DWMAC1000
[    4.567742] rk_gmac-dwmac ff540000.ethernet: DMA HW capability register supported
[    4.567756] rk_gmac-dwmac ff540000.ethernet: RX Checksum Offload Engine supported
[    4.567768] rk_gmac-dwmac ff540000.ethernet: COE Type 2
[    4.567782] rk_gmac-dwmac ff540000.ethernet: TX Checksum insertion supported
[    4.567794] rk_gmac-dwmac ff540000.ethernet: Wake-Up On Lan supported
[    4.567960] rk_gmac-dwmac ff540000.ethernet: Normal descriptors
[    4.567984] rk_gmac-dwmac ff540000.ethernet: Ring mode enabled
[    4.567996] rk_gmac-dwmac ff540000.ethernet: Enable RX Mitigation via HW Watchdog Timer
[    4.682397] rk_gmac-dwmac ff540000.ethernet end0: renamed from eth0
[    5.707704] usb 3-1: new SuperSpeed USB device number 2 using xhci-hcd
[    5.717309] EXT4-fs (mmcblk0p1): mounted filesystem ed18e1a3-fb1e-4044-9e49-1397fba1aa37 ro with writeback data mode. Quota mode: none.
[    5.726023] usb 3-1: New USB device found, idVendor=0bda, idProduct=8153, bcdDevice=31.00
[    5.726060] usb 3-1: New USB device strings: Mfr=1, Product=2, SerialNumber=6
[    5.726078] usb 3-1: Product: USB 10/100/1000 LAN
[    5.726092] usb 3-1: Manufacturer: Realtek
[    5.726105] usb 3-1: SerialNumber: 000000000000
[    5.817251] usbcore: registered new device driver r8152-cfgselector
[    5.994893] r8152-cfgselector 3-1: reset SuperSpeed USB device number 2 using xhci-hcd
[    6.083774] r8152 3-1:1.0 eth0: v1.12.13
[    6.083958] usbcore: registered new interface driver r8152
[    6.089196] usbcore: registered new interface driver cdc_ether
[    6.634061] systemd[1]: Inserted module 'autofs4'
[    6.693724] systemd[1]: systemd 252.33-1~deb12u1 running in system mode (+PAM +AUDIT +SELINUX +APPARMOR +IMA +SMACK +SECCOMP +GCRYPT -GNUTLS +OPENSSL +ACL +BLKID +CURL +ELFUTILS +FIDO2 +IDN2 -IDN +IPTC +KMOD +LIBCRYPTSETUP +LIBFDISK +PCRE2 -PWQUALITY +P11KIT +QRENCODE +TPM2 +BZIP2 +LZ4 +XZ +ZLIB +ZSTD -BPF_FRAMEWORK -XKBCOMMON +UTMP +SYSVINIT default-hierarchy=unified)
[    6.693783] systemd[1]: Detected architecture arm64.
[    6.701481] systemd[1]: Hostname set to <nanopi-r2s>.
[    7.252508] (sd-execut[178]: /usr/lib/systemd/system-generators/netplan failed with exit status 1.
[    7.620694] systemd[1]: /etc/systemd/system/vlmcsd.service:7: Special user nobody configured, this is not safe!
[    8.035822] systemd[1]: Queued start job for default target graphical.target.
[    8.083423] systemd[1]: Created slice system-getty.slice - Slice /system/getty.
[    8.087128] systemd[1]: Created slice system-modprobe.slice - Slice /system/modprobe.
[    8.090495] systemd[1]: Created slice system-serial\x2dgetty.slice - Slice /system/serial-getty.
[    8.093025] systemd[1]: Created slice user.slice - User and Session Slice.
[    8.093867] systemd[1]: Started systemd-ask-password-wall.path - Forward Password Requests to Wall Directory Watch.
[    8.095600] systemd[1]: Set up automount proc-sys-fs-binfmt_misc.automount - Arbitrary Executable File Formats File System Automount Point.
[    8.096084] systemd[1]: Expecting device dev-ttyGS0.device - /dev/ttyGS0...
[    8.096364] systemd[1]: Expecting device dev-ttyS2.device - /dev/ttyS2...
[    8.096875] systemd[1]: Reached target integritysetup.target - Local Integrity Protected Volumes.
[    8.097688] systemd[1]: Reached target slices.target - Slice Units.
[    8.098159] systemd[1]: Reached target swap.target - Swaps.
[    8.098553] systemd[1]: Reached target time-set.target - System Time Set.
[    8.099159] systemd[1]: Reached target veritysetup.target - Local Verity Protected Volumes.
[    8.129034] systemd[1]: Listening on rpcbind.socket - RPCbind Server Activation Socket.
[    8.130553] systemd[1]: Listening on syslog.socket - Syslog Socket.
[    8.132045] systemd[1]: Listening on systemd-fsckd.socket - fsck to fsckd communication Socket.
[    8.132997] systemd[1]: Listening on systemd-initctl.socket - initctl Compatibility Named Pipe.
[    8.137837] systemd[1]: Listening on systemd-journald-audit.socket - Journal Audit Socket.
[    8.139191] systemd[1]: Listening on systemd-journald-dev-log.socket - Journal Socket (/dev/log).
[    8.140838] systemd[1]: Listening on systemd-journald.socket - Journal Socket.
[    8.143057] systemd[1]: Listening on systemd-udevd-control.socket - udev Control Socket.
[    8.144367] systemd[1]: Listening on systemd-udevd-kernel.socket - udev Kernel Socket.
[    8.151663] systemd[1]: Mounting dev-hugepages.mount - Huge Pages File System...
[    8.159814] systemd[1]: Mounting dev-mqueue.mount - POSIX Message Queue File System...
[    8.168051] systemd[1]: Mounting sys-kernel-debug.mount - Kernel Debug File System...
[    8.176353] systemd[1]: Mounting sys-kernel-tracing.mount - Kernel Trace File System...
[    8.177717] systemd[1]: auth-rpcgss-module.service - Kernel Module supporting RPCSEC_GSS was skipped because of an unmet condition check (ConditionPathExists=/etc/krb5.keytab).
[    8.187792] systemd[1]: Starting fake-hwclock.service - Restore / save the current clock...
[    8.197120] systemd[1]: Starting keyboard-setup.service - Set the console keyboard layout...
[    8.206708] systemd[1]: Starting kmod-static-nodes.service - Create List of Static Device Nodes...
[    8.216595] systemd[1]: Starting modprobe@configfs.service - Load Kernel Module configfs...
[    8.227589] systemd[1]: Starting modprobe@dm_mod.service - Load Kernel Module dm_mod...
[    8.239218] systemd[1]: Starting modprobe@drm.service - Load Kernel Module drm...
[    8.250633] systemd[1]: Starting modprobe@efi_pstore.service - Load Kernel Module efi_pstore...
[    8.262317] systemd[1]: Starting modprobe@fuse.service - Load Kernel Module fuse...
[    8.272235] systemd[1]: Starting modprobe@loop.service - Load Kernel Module loop...
[    8.276011] systemd[1]: systemd-fsck-root.service - File System Check on Root Device was skipped because of an unmet condition check (ConditionPathExists=!/run/initramfs/fsck-root).
[    8.291103] systemd[1]: Starting systemd-modules-load.service - Load Kernel Modules...
[    8.299971] device-mapper: uevent: version 1.0.3
[    8.300827] device-mapper: ioctl: 4.48.0-ioctl (2023-03-01) initialised: dm-devel@lists.linux.dev
[    8.301008] systemd[1]: Starting systemd-remount-fs.service - Remount Root and Kernel File Systems...
[    8.311096] systemd[1]: Starting systemd-udev-trigger.service - Coldplug All udev Devices...
[    8.330347] systemd[1]: Mounted dev-hugepages.mount - Huge Pages File System.
[    8.333422] systemd[1]: Mounted dev-mqueue.mount - POSIX Message Queue File System.
[    8.337023] systemd[1]: Mounted sys-kernel-debug.mount - Kernel Debug File System.
[    8.340922] systemd[1]: Mounted sys-kernel-tracing.mount - Kernel Trace File System.
[    8.344867] systemd[1]: Finished fake-hwclock.service - Restore / save the current clock.
[    8.349493] systemd[1]: Finished kmod-static-nodes.service - Create List of Static Device Nodes.
[    8.354322] systemd[1]: modprobe@configfs.service: Deactivated successfully.
[    8.355845] systemd[1]: Finished modprobe@configfs.service - Load Kernel Module configfs.
[    8.359028] systemd[1]: modprobe@dm_mod.service: Deactivated successfully.
[    8.360777] systemd[1]: Finished modprobe@dm_mod.service - Load Kernel Module dm_mod.
[    8.364185] systemd[1]: modprobe@efi_pstore.service: Deactivated successfully.
[    8.366041] systemd[1]: Finished modprobe@efi_pstore.service - Load Kernel Module efi_pstore.
[    8.370268] systemd[1]: modprobe@drm.service: Deactivated successfully.
[    8.372774] systemd[1]: Finished modprobe@drm.service - Load Kernel Module drm.
[    8.375591] systemd[1]: modprobe@fuse.service: Deactivated successfully.
[    8.377591] systemd[1]: Finished modprobe@fuse.service - Load Kernel Module fuse.
[    8.381290] systemd[1]: modprobe@loop.service: Deactivated successfully.
[    8.382873] systemd[1]: Finished modprobe@loop.service - Load Kernel Module loop.
[    8.420229] systemd[1]: Mounting sys-fs-fuse-connections.mount - FUSE Control File System...
[    8.430215] systemd[1]: Mounting sys-kernel-config.mount - Kernel Configuration File System...
[    8.431731] g_serial gadget.0: Gadget Serial v2.4
[    8.431761] g_serial gadget.0: g_serial ready
[    8.431779] dwc2 ff580000.usb: bound driver g_serial
[    8.432069] systemd[1]: systemd-repart.service - Repartition Root Disk was skipped because no trigger condition checks were met.
[    8.435328] EXT4-fs (mmcblk0p1): re-mounted ed18e1a3-fb1e-4044-9e49-1397fba1aa37 r/w. Quota mode: none.
[    8.446387] dwc2 ff580000.usb: dwc2_core_reset: HANG! Soft Reset timeout GRSTCTL_CSFTRST
[    8.464786] systemd[1]: Finished systemd-modules-load.service - Load Kernel Modules.
[    8.468623] systemd[1]: Finished systemd-remount-fs.service - Remount Root and Kernel File Systems.
[    8.471453] systemd[1]: Mounted sys-fs-fuse-connections.mount - FUSE Control File System.
[    8.473750] systemd[1]: Mounted sys-kernel-config.mount - Kernel Configuration File System.
[    8.476690] systemd[1]: systemd-pstore.service - Platform Persistent Storage Archival was skipped because of an unmet condition check (ConditionDirectoryNotEmpty=/sys/fs/pstore).
[    8.508542] systemd[1]: Starting systemd-random-seed.service - Load/Save Random Seed...
[    8.517354] systemd[1]: Starting systemd-sysctl.service - Apply Kernel Variables...
[    8.527894] systemd[1]: Starting systemd-sysusers.service - Create System Users...
[    8.641156] systemd[1]: Finished systemd-sysctl.service - Apply Kernel Variables.
[    8.651653] systemd[1]: Finished systemd-sysusers.service - Create System Users.
[    8.673028] systemd[1]: Starting systemd-tmpfiles-setup-dev.service - Create Static Device Nodes in /dev...
[    8.788053] systemd[1]: Finished systemd-tmpfiles-setup-dev.service - Create Static Device Nodes in /dev.
[    8.824695] systemd[1]: Starting systemd-udevd.service - Rule-based Manager for Device Events and Files...
[    8.878679] systemd[1]: Finished keyboard-setup.service - Set the console keyboard layout.
[    8.881624] systemd[1]: Reached target local-fs-pre.target - Preparation for Local File Systems.
[    9.064584] systemd[1]: Finished systemd-udev-trigger.service - Coldplug All udev Devices.
[    9.152678] systemd[1]: Starting ifupdown-pre.service - Helper to synchronize boot up for ifupdown...
[    9.173688] systemd[1]: Finished ifupdown-pre.service - Helper to synchronize boot up for ifupdown.
[    9.251316] systemd[1]: Started systemd-udevd.service - Rule-based Manager for Device Events and Files.
[    9.288169] systemd[1]: Starting plymouth-start.service - Show Plymouth Boot Screen...
[    9.305756] systemd[1]: Found device dev-ttyGS0.device - /dev/ttyGS0.
[    9.385454] systemd[1]: Mounting tmp.mount - /tmp...
[    9.398678] systemd[1]: Started plymouth-start.service - Show Plymouth Boot Screen.
[    9.405226] systemd[1]: systemd-ask-password-console.path - Dispatch Password Requests to Console Directory Watch was skipped because of an unmet condition check (ConditionPathExists=!/run/plymouth/pid).
[    9.406116] systemd[1]: Started systemd-ask-password-plymouth.path - Forward Password Requests to Plymouth Directory Watch.
[    9.406831] systemd[1]: Reached target cryptsetup.target - Local Encrypted Volumes.
[    9.407262] systemd[1]: Reached target paths.target - Path Units.
[    9.420729] systemd[1]: Mounted tmp.mount - /tmp.
[    9.426131] systemd[1]: Reached target local-fs.target - Local File Systems.
[    9.434719] systemd[1]: Starting armbian-zram-config.service - Armbian ZRAM config...
[    9.446346] systemd[1]: Starting console-setup.service - Set console font and keymap...
[    9.461045] systemd[1]: Starting plymouth-read-write.service - Tell Plymouth To Write Out Runtime Data...
[    9.496172] systemd[1]: Starting systemd-binfmt.service - Set Up Additional Binary Formats...
[    9.504643] systemd[1]: Finished console-setup.service - Set console font and keymap.
[    9.506909] systemd[1]: Received SIGRTMIN+20 from PID 143 (plymouthd).
[    9.535802] systemd[1]: Found device dev-ttyS2.device - /dev/ttyS2.
[    9.541858] systemd[1]: Finished plymouth-read-write.service - Tell Plymouth To Write Out Runtime Data.
[    9.595042] systemd[1]: proc-sys-fs-binfmt_misc.automount: Got automount request for /proc/sys/fs/binfmt_misc, triggered by 272 (systemd-binfmt)
[    9.635716] systemd[1]: Mounting proc-sys-fs-binfmt_misc.mount - Arbitrary Executable File Formats File System...
[    9.694231] systemd[1]: Mounted proc-sys-fs-binfmt_misc.mount - Arbitrary Executable File Formats File System.
[    9.707994] systemd[1]: Finished systemd-binfmt.service - Set Up Additional Binary Formats.
[    9.897927] zram: Added device: zram0
[    9.900383] zram: Added device: zram1
[    9.955026] zram: Added device: zram2
[    9.964391] r8152-cfgselector 3-1: reset SuperSpeed USB device number 2 using xhci-hcd
[   10.106034] zram0: detected capacity change from 0 to 993224
[   10.120358] r8152 3-1:1.0 eth0: v1.12.13
[   10.139091] rk3288-crypto ff060000.crypto: will run requests pump with realtime priority
[   10.139141] rk3288-crypto ff060000.crypto: Register ecb(aes) as ecb-aes-rk
[   10.139191] rk3288-crypto ff060000.crypto: Register cbc(aes) as cbc-aes-rk
[   10.139208] rk3288-crypto ff060000.crypto: Register ecb(des) as ecb-des-rk
[   10.139224] rk3288-crypto ff060000.crypto: Register cbc(des) as cbc-des-rk
[   10.139240] rk3288-crypto ff060000.crypto: Register ecb(des3_ede) as ecb-des3-ede-rk
[   10.139257] rk3288-crypto ff060000.crypto: Register cbc(des3_ede) as cbc-des3-ede-rk
[   10.139273] rk3288-crypto ff060000.crypto: Register sha1 as rk-sha1
[   10.139291] rk3288-crypto ff060000.crypto: Register sha256 as rk-sha256
[   10.139308] rk3288-crypto ff060000.crypto: Register md5 as rk-md5
[   10.294349] systemd[1]: Reached target usb-gadget.target - Hardware activated USB gadget.
[   10.305461] mc: Linux media interface: v0.10
[   10.316121] input: rk805 pwrkey as /devices/platform/ff160000.i2c/i2c-1/1-0018/rk805-pwrkey.5.auto/input/input1
[   10.359334] videodev: Linux video capture interface: v2.00
[   10.445710] rockchip-rga ff390000.rga: HW Version: 0x04.00
[   10.446343] rockchip-rga ff390000.rga: Registered rockchip-rga as /dev/video0
[   10.641905] rockchip_vdec: module is from the staging directory, the quality is unknown, you have been warned.
[   10.654058] rkvdec ff360000.video-codec: Adding to iommu group 0
[   10.794714] r8152 3-1:1.0 enxfa8db878cde4: renamed from eth0
[   13.147727] Adding 496608k swap on /dev/zram0.  Priority:5 extents:1 across:496608k SS
[   13.193975] zram1: detected capacity change from 0 to 102400
[   13.395498] random: crng init done
[   13.406104] systemd[1]: Finished systemd-random-seed.service - Load/Save Random Seed.
[   13.408229] systemd[1]: first-boot-complete.target - First Boot Complete was skipped because of an unmet condition check (ConditionFirstBoot=yes).
[   13.408518] systemd[1]: systemd-machine-id-commit.service - Commit a transient machine-id on disk was skipped because of an unmet condition check (ConditionPathIsMountPoint=/etc/machine-id).
[   13.500047] systemd[1]: Finished armbian-zram-config.service - Armbian ZRAM config.
[   13.529266] systemd[1]: Starting armbian-ramlog.service - Armbian memory supported logging...
[   13.598885] EXT4-fs (zram1): mounted filesystem 75d615b2-8bfa-48d0-8bee-f23dbc080c49 r/w without journal. Quota mode: none.
[   14.690872] systemd[1]: Finished armbian-ramlog.service - Armbian memory supported logging.
[   14.741090] systemd[1]: Starting systemd-journald.service - Journal Service...
[   14.945933] systemd[1]: Started systemd-journald.service - Journal Service.
[   15.005232] systemd-journald[405]: Received client request to flush runtime journal.
[   15.065380] systemd-journald[405]: Data hash table of /var/log/journal/db8076f9fe0a450ab450ce46ac0c4061/system.journal has a fill level at 84.5 (3847 of 4551 items, 2621440 file size, 681 bytes per hash table item), suggesting rotation.
[   15.065412] systemd-journald[405]: /var/log/journal/db8076f9fe0a450ab450ce46ac0c4061/system.journal: Journal header limits reached or header out-of-date, rotating.
[   15.473394] RPC: Registered named UNIX socket transport module.
[   15.473418] RPC: Registered udp transport module.
[   15.473422] RPC: Registered tcp transport module.
[   15.473427] RPC: Registered tcp-with-tls transport module.
[   15.473431] RPC: Registered tcp NFSv4.1 backchannel transport module.
[   16.386707] systemd-journald[405]: Received client request to relinquish /var/log/journal/db8076f9fe0a450ab450ce46ac0c4061 access.
[   19.609028] rk_gmac-dwmac ff540000.ethernet end0: Register MEM_TYPE_PAGE_POOL RxQ-0
[   19.620809] rk_gmac-dwmac ff540000.ethernet end0: PHY [stmmac-0:03] driver [YT8521 Gigabit Ethernet] (irq=POLL)
[   19.627726] rk_gmac-dwmac ff540000.ethernet end0: No Safety Features support found
[   19.627763] rk_gmac-dwmac ff540000.ethernet end0: PTP not supported by HW
[   19.634301] rk_gmac-dwmac ff540000.ethernet end0: configuring for phy/rgmii link mode
[   19.712283] r8152 3-1:1.0 enxfa8db878cde4: carrier on
[   26.780550] systemd-journald[405]: Time jumped backwards, rotating.
[  182.437747] rk_gmac-dwmac ff540000.ethernet end0: Link is Up - 1Gbps/Full - flow control off
